lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1732,6 +1740,26 @@ def set_log_entry_fields(record, entry)
       end
     end
 
+    # Parse labels. Return nil if not set.
+    def parse_labels(record)
+      payload_labels = record.delete(@labels_key)
+      unless payload_labels.is_a?(Hash)
+        @log.error "Invalid value of '#{@labels_key}' in the payload: " \
+                   "#{payload_labels}. Labels need to be a JSON object.", err
+        return nil
+      end
+      payload_labels.each do |k, v|
+        next if k.is_a?(String) && v.is_a?(String)
+        @log.error "Invalid value of '#{@labels_key}' in the payload. " \
```

Comment:
Was this supposed to end in a ':' instead of a '.'?

Reply:
Yup. Changed.